src/process/sys/linux/process.rs

```diff
@@ -120,45 +119,33 @@ impl Process {
 		todo!()
 	}
 
-	pub(crate) fn sys_chidren(&self) {
+	pub(crate) fn sys_children(&self) {
 		todo!()
 	}
 
 	pub(crate) fn sys_open_files(&self) -> ProcessResult<Vec<OpenFile>> {
 		read_dir(self.procfs_path("fd"))
 			.map_err(|e| psutil_error_to_process_error(e, self.pid))?
 			.into_iter()
-			.map(|entry| {
+			.filter_map(|entry| {
 				let path = entry.path();
-				// TODO: fix or document the unwraps
-				let fd = path
-					.file_name()
-					.unwrap()
-					.to_string_lossy()
-					.parse::<u32>()
-					.unwrap();
-				let open_file =
-					read_link(&path).map_err(|e| psutil_error_to_process_error(e, self.pid))?;
-
-				Ok(OpenFile {
-					fd: Some(fd),
-					path: open_file,
-				})
+
+				Some(Ok(OpenFile {
+					fd: Some(
+						path.file_name()
+							.expect("unable to decode fd")
+							.to_string_lossy()
+							.parse::<u32>()
+							.ok()?,
+					),
+					path: match read_link(&path) {
+						Ok(path) => path,
+						Err(e) => return Some(Err(psutil_error_to_process_error(e, self.pid))),
+					},
+				}))
 			})
 			.collect()
 	}
-
-	pub(crate) fn sys_connections(&self) {
```

Comment:
I think we're going to need these in the future.

Reply:
ok, reverted.